crates/database/src/quotes.rs

```diff
@@ -136,12 +140,88 @@ pub async fn remove_expired_quotes(
     const QUERY: &str = r#"
 DELETE FROM quotes
 WHERE expiration_timestamp < $1
+RETURNING id
     "#;
-    sqlx::query(QUERY)
+    let deleted_ids = sqlx::query(QUERY)
         .bind(max_expiry)
+        .fetch_all(&mut *ex)
+        .await?
+        .iter()
+        .map(|row| row.get(0))
+        .collect::<Vec<_>>();
+
+    delete_quote_interactions(ex, &deleted_ids).await
```

Comment:
This is not executed within the same transaction. It should be fixed or consider using the foreign key, as suggested above.